### PR TITLE
Add <th> with scope='row' to Table component for screen readers

### PR DESCRIPTION
**Pull request by ddzz**
## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/489

## Testing done
Manual testing.

## Acceptance criteria
- [ ] Table component rows appear as described in the ticket.

### CODE REVIEW

**Review by CBonade — APPROVED** [review]

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -48,20 +48,36 @@ function Table(props) {
...
+                <th
+                  data-index={index}
+                  className={classNames(borderClasses, {
+                    'vads-u-text-align--left': field.alignLeft,
+                    'medium-screen:vads-u-text-align--right': field.alignRight,
+                  })}
+                  data-label={field.label}
+                  key={`${rowIndex}-${field.label}`}
+                  role="cell"
+                  scope="row"
```
Comment (mpelzsherman) [inline src/components/Table/Table.jsx:62]: There's a lot of repetition here.  What do you think about creating an object with the common key/value pairs, and using the spread operator to add to each element? e.g.

```jsx
const cellProps = {
  'data-index': { index },
  className: {
    classNames(borderClasses, {
      'vads-u-text-align--left': field.alignLeft,
      'medium-screen:vads-u-text-align--right': field.alignRight,
    }),
  },
  'data-label': { field.label },
  ...etc.
};

const thProps = {
  ...cellProps,
  scope: 'row',
}


{fields.map((field, index) =>
  index === 0 ? (
    <th {...thProps} />
  ) : (
    <td {...cellProps} />
```

Comment (ddzz) [reply src/components/Table/Table.jsx]: Thanks for reviewing! I updated the PR with this approach. I had to use functions to create the objects.

**Review by mpelzsherman — APPROVED** [review]

**Review by cvalarida — COMMENTED** [review]
The change to the table component looks good, but can you add a test to check that the first element in a `tr` is a `th` with `scope="row"`?

Also, I'm really confused about the change to that existing test. :sweat_smile:

File: src/components/Table/Table.unit.spec.jsx [hunk 2]
```diff
@@ -35,7 +35,8 @@ describe('<Table />', () => {
     );
 
     expect(wrapper.find('thead')).to.have.lengthOf(1);
-    expect(wrapper.find('th')).to.have.lengthOf(6); // Once for each field
+    expect(wrapper.find('th')).to.have.lengthOf(15);
+    expect(wrapper.find('td')).to.have.lengthOf(45);
```
[the remark this answers is not in the document]
Comment (ddzz) [reply src/components/Table/Table.unit.spec.jsx]: There are 60 total cells in the table based on the test data.

**Comment by ddzz** [comment]
> The change to the table component looks good, but can you add a test to check that the first element in a `tr` is a `th` with `scope="row"`?
> 
> Also, I'm really confused about the change to that existing test. 😅

Thanks for the review! Specs have been updated.

**Review by cvalarida — APPROVED** [review]
Thanks for the updated tests! :shipit:

**Pull request merged**